pyprep/find_noisy_channels.py

```diff
@@ -366,23 +366,25 @@ def find_bad_by_correlation(
         for k in range(0, w_correlation):
             eeg_portion = np.transpose(np.squeeze(EEG_new_win[:, :, k]))
             data_portion = np.transpose(np.squeeze(data_win[:, :, k]))
-            window_correlation = np.corrcoef(np.transpose(eeg_portion))
+            with np.errstate(invalid='ignore'):  # suppress divide-by-zero warnings
+                window_correlation = np.corrcoef(np.transpose(eeg_portion))
```

Comment:
in which cases would the `np.errstate(invalid='ignore')` be necessary? And could it results in `window_correlation` being NaN?

Reply:
Yep, inside `np.corrcoef` it uses the standard deviation of the inputs at some point in the calculation, and since for a dropout channel the SD is going to be 0, we get NaNs because of the division-by-zero error (and a corresponding RuntimeWarning message). However, the current code uses those NaNs to determine which channels are dropout channels for each window so things would need to be refactored a fair bit to prevent the NaNs in the first place.

I'm thinking of doing a proper refactor of NoisyChannels at some point once MATLAB comparison is merged, so I went with the path of least resistance for now.

Reply:
To clarify, the `with np.errstate` is just to suppress some division-by-zero runtime warnings that are currently inevitable/expected based on how the code is structured.

Reply:
that makes sense, thanks for clarifying.